src/lib/socket.c

```diff
@@ -57,6 +57,42 @@
 #include <openssl/ssl.h>
 #include <openssl/err.h>
 #include <openssl/bio.h>
+#if OPENSSL_VERSION_NUMBER < 0x10100001L
```

Comment:
Have libressl talked about what they're going to do then? 
OpenSSL has indicated that this is the recommended way to keep backwards compat. e.g. see https://wiki.openssl.org/index.php/1.1_API_Changes#Adding_forward-compatible_code_to_older_versions
This is same way that I recommend keeping lua compat (i.e. code for the newest version; add backwards compatibility wrappers), so I won't disagree with them.